addons/obiaway/Obiaway.lua

```diff
@@ -150,36 +314,232 @@ function get_all_elements()
+            windower.send_command('put "%s Obi" %s;':format(name, settings.location))
+            if settings.notify then
+                obi_output('Putting %s Obi away into %s.':format(name, settings.location))
+            end
+            wait(.5)
+        end
+    end
+
+    return 1
+end
+
+function get_all_obi(command)
+    if inventory_full(command) then return 0 end
+    local obi = get_obi_in_inventory()
+    local obi_bag = get_obi_in_inventory(settings.location)
+    if free_space() < obi_bag["n"] then
+        obi_output('Not enough space in inventory...')
+        return 0
+    end
+
+    local elements = get_all_elements()
+    local obi = get_obi_in_inventory()
+
+    for name, element in obi_names:it() do
+        if not obi[element] then
+            windower.send_command('get "%s Obi" %s;':format(name, settings.location))
+            if settings.notify then
+                obi_output('Getting %s Obi from %s.':format(name, settings.location))
+            end
+            wait(.5)
+        end
+    end
+    
+    return 1
+end
+
+function put_all_obi(command)
+    if inventory_full(command, settings.location) then return 0 end
+    local obi = get_obi_in_inventory()
+    if free_space(settings.location) < obi["n"] then
+        obi_output('Not enough space in %s...':format(settings.location))
+        return 0
+    end
+
+    local elements = get_all_elements()
+
+    for name, element in obi_names:it() do
+        if obi[element] then
+            windower.send_command('put "%s Obi" %s;':format(name, settings.location))
+            if settings.notify then
+                obi_output('Putting %s Obi away into %s.':format(name, settings.location))
+            end
+            wait(.5)
+        end
+    end
+
+    return 1
+end
 
-windower.register_event('day change', 'weather change', remove_unneeded_obis)
+-- function called on automatic events. sorts obi based on location.
+function auto_sort_obi()
+    -- if inventory and obi bag are full at the same time, do nothing. 'cause we can't.
+    if inventory_full(false) and inventory_full(false, settings.location) then return end
 
+    if not settings.lock then -- if sorting lock is not on, then do this stuff:
+        if not obi_cities:contains(res.zones[windower.ffxi.get_info().zone].english) then -- In a city:
+            put_unneeded_obi(false)
+            get_needed_obi(false)
+        else-- Not in a city:
+            put_all_obi(false)
+        end
+    end
+end
+
+----------------------------------------
+---            EVENTS
+----------------------------------------
 
+windower.register_event('gain buff', auto_sort_obi:cond(function(id) return id >= 178 and id <= 185 end))
+windower.register_event('lose buff', auto_sort_obi:cond(function(id) return id >= 178 and id <= 185 end))
+windower.register_event('day change', 'weather change', auto_sort_obi)
+windower.register_event('addon command', function(command, ...)
+    local command = command:lower() or 'help'
```

Comment:
The `local` is redundant, since it's already function-local (all arguments are). Also, this will error if someone tries to call it without any arguments. You want this:
```lua
    command = command and command:lower() or 'help'
```